component/discovery/kuma/kuma.go

```diff
@@ -0,0 +1,77 @@
+package kuma
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/common/config"
+	"github.com/grafana/agent/component/discovery"
+	"github.com/prometheus/common/model"
+	prom_discovery "github.com/prometheus/prometheus/discovery/xds"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "discovery.kuma",
+		Args:    Arguments{},
+		Exports: discovery.Exports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+// Arguments configure the discovery.kuma component.
+type Arguments struct {
+	Server          string        `river:"server,attr"`
+	RefreshInterval time.Duration `river:"refresh_interval,attr,optional"`
+	FetchTimeout    time.Duration `river:"fetch_timeout,attr,optional"`
+
+	HTTPClientConfig config.HTTPClientConfig `river:",squash"`
+}
+
+// DefaultArguments is used to initialize default values for Arguments.
+var DefaultArguments = Arguments{
+	RefreshInterval: 30 * time.Second,
+	FetchTimeout:    2 * time.Minute,
+
+	HTTPClientConfig: config.DefaultHTTPClientConfig,
+}
+
+// SetToDefault implements river.Defaulter.
+func (args *Arguments) SetToDefault() {
+	*args = DefaultArguments
+}
+
+// Validate implements river.Validator.
+func (args *Arguments) Validate() error {
+	if args.RefreshInterval <= 0 {
+		return fmt.Errorf("refresh_interval must be greater than 0")
+	}
+	if args.FetchTimeout <= 0 {
+		return fmt.Errorf("fetch_timeout must be greater than 0")
+	}
+
+	return args.HTTPClientConfig.Validate()
+}
+
+// Convert returns the upstream configuration struct.
+func (args *Arguments) Convert() *prom_discovery.SDConfig {
```

Comment:
I think that both Validate and Convert should be covered by tests to make sure that the data is passed to the prometheus code correctly

Reply:
I've added a couple of rudimentary tests, although I don't believe they offer much here.

Reply:
Last time I forgot to convert the TLS config and the code compiled and everything seemed fine. But later when I made the converters I realised that if I did not call the convert function, the TLS config was just ignored. (no error, it was just not mapped). Besides I agree that it does not bring so much here, it's hard to find the right balance and I don't know what would work best for this project